File: 1014_Best Sightseeing Pair/BestSightseeingPair.cs
public class Solution {
    public int MaxScoreSightseeingPair(int[] values) {
        var currMax = values[0];
        var ans = 0;
        
        for(int j=1;j<values.Length;j++)
        {
            ans = Math.Max(ans, currMax + values[j] - j);
            currMax = Math.Max(currMax, values[j] + j);
        }
        
        return ans;
    }
}

File: 1014_Best Sightseeing Pair/BestSightseeingPair.java
class Solution {
    public int maxScoreSightseeingPair(int[] values) {
        int ans = 0;
        int currMax = values[0];
        
        for(int j=1;j<values.length;j++) {
            ans = Math.max(ans, currMax + values[j] - j);
            currMax = Math.max(currMax, values[j] + j);
        }
        
        return ans;
    }
}